packages/actor-core/src/actor/schedule.ts

```diff
@@ -28,27 +37,109 @@ export class Schedule {
 		this.#driver = driver;
 	}
 
-	async after(duration: number, fn: string, ...args: unknown[]) {
-		await this.#scheduleEvent(Date.now() + duration, fn, args);
+	async after(duration: number, fn: string, ...args: unknown[]): Promise<string> {
+		return this.#scheduleEvent(Date.now() + duration, fn, args);
+	}
+
+	async at(timestamp: number, fn: string, ...args: unknown[]): Promise<string> {
+		return this.#scheduleEvent(timestamp, fn, args);
+	}
+
+	async get(alarmId: string): Promise<Alarm | null> {
+		const event = await this.#driver.kvGet(
+			this.#actor.id,
+			KEYS.SCHEDULE.event(alarmId)
+		) as ScheduleEvent | undefined;
+
+		if (!event) return null;
+
+		return {
+			id: alarmId,
+			createdAt: event.createdAt,
+			triggersAt: event.timestamp,
+			fn: event.fn,
+			args: event.args
+		};
+	}
+
+	async list(): Promise<readonly Alarm[]> {
+		const schedule: ScheduleState = ((await this.#driver.kvGet(
+			this.#actor.id,
+			KEYS.SCHEDULE.SCHEDULE
+		)) as ScheduleState) ?? { events: [] };
+
+		const alarms: Alarm[] = [];
+		for (const event of schedule.events) {
+			const scheduleEvent = await this.#driver.kvGet(
+				this.#actor.id,
+				KEYS.SCHEDULE.event(event.eventId)
+			) as ScheduleEvent;
+
+			if (scheduleEvent) {
+				alarms.push(Object.freeze({
+					id: event.eventId,
+					createdAt: scheduleEvent.createdAt,
+					triggersAt: scheduleEvent.timestamp,
+					fn: scheduleEvent.fn,
+					args: scheduleEvent.args
+				}));
+			}
+		}
+
+		return Object.freeze(alarms);
```

Comment:
The `Object.freeze` here and above is just to ensure the alarm data structure can't be mutated. Probably should do the same for `.get`